specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json

```diff
@@ -0,0 +1,1023 @@
+            "description": "Capacity tag value"
+          }
+        },
+        "applicationPorts": {
+          "$ref": "#/definitions/EndpointRangeDescription",
+          "description": "The range of ports from which cluster assigned port to Service Fabric applications."
+        },
+        "ephemeralPorts": {
+          "$ref": "#/definitions/EndpointRangeDescription",
+          "description": "The range of ephemeral ports that nodes in this node type should be configured with."
+        },
+        "vmSecrets": {
+          "type": "array",
+          "title": "virtual machine secretes.",
+          "description": "The secrets to install in the virtual machines.",
+          "items": {
+            "$ref": "#/definitions/VaultSecretGroup"
+          }
+        },
+        "vmExtensions": {
+          "type": "array",
+          "title": "virtual machine extensions.",
+          "description": "Set of extensions that should be installed onto the virtual machines.",
+          "items": {
+            "$ref": "#/definitions/VMSSExtension"
+          }
+        }
+      },
+      "description": "Node type update properties."
+    },
+    "NodeTypeUpdateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NodeTypePropertiesUpdateParameters",
+          "description": "Node type update properties."
+        },
+        "tags": {
+          "type": "object",
+          "description": "Node type update parameters",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      },
+      "description": "Node type update request"
+    },
+    "ProxyResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Azure resource identifier.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Azure resource type.",
+          "readOnly": true
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Azure resource tags.",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        }
+      },
+      "description": "The resource model definition for proxy-only resource.",
+      "x-ms-azure-resource": true
+    },
+    "SubResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Azure resource identifier."
+        }
+      },
+      "description": "Azure resource identifier.",
+      "x-ms-azure-resource": true
+    },
+    "VaultCertificate": {
+      "required": [
+        "certificateUrl",
+        "certificateStore"
+      ],
+      "properties": {
+        "certificateUrl": {
```

Comment:
>certificateUrl [](start = 9, length = 14)

Please go through [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=best%20practice](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=best%20practice)
In this case please annotate using x-ms-secret: true

Reply:
I'm not adding it here, we are doing the same as VMSS where the certificates are returned in the get. https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json#L8449

Reply:
This may have been missed by manual reviewer. 
Since this will be returned in the GET response its important for other components like Deployments and AzureResourceGraph to make sure this data is not stored and shown back to the user in these components.

Reply:
this is not the actual secret its just the identifier in the key vault why can't it be returned in the get?